pkg/internal/gomod/repo.go

```diff
@@ -0,0 +1,39 @@
+package gomod
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"go.opentelemetry.io/otel"
+)
+
+// Repo returns the source code repository URL for the given module import path.
+func Repo(ctx context.Context, importpath string) (string, error) {
+	_, span := otel.Tracer("wolfictl").Start(ctx, fmt.Sprintf("resolving git URL for go module %s", importpath))
+	defer span.End()
+
+	if strings.HasPrefix(importpath, "github.com/") {
```

Comment:
This only matches github.com but we have certain projects using gitlab.com. Let's add both use cases.